files/var/www/ks/bootstrap_simp_client

```diff
@@ -0,0 +1,564 @@
+        'script is allowed to run.  Script will',
+        'abort if it does not complete within',
+        "this allotted time. Defaults to #{DEFAULT_MAX_SECONDS}."
+      ) do |max_seconds|
+        @options[:max_seconds] = max_seconds
+      end
+
+      opts.on(
+        '-S', '--[no-]print-stats',
+        'Print statistics for each puppet run',
+        "during bootstrap. Defaults to #{DEFAULT_PRINT_STATS}."
+      ) do |print_stats|
+        @options[:print_stats] = print_stats
+      end
+
+      opts.on(
+        '-w', '--puppet-wait-for-cert WAITSECONDS',
+        Integer,
+        'The wait interval in seconds for checking',
+        'into the puppet master for a signed',
+        'certificate. This checking will only',
+        'continue until a signed certificate is',
+        'presented.  If set to 0, each puppet agent',
+        'run will fail if a signed certificate is',
+        "not presented. Defaults to #{DEFAULT_PUPPET_WAIT_FOR_CERT}."
+      ) do |puppet_wait_for_cert|
+        @options[:puppet_wait_for_cert] = puppet_wait_for_cert
+      end
+
+      opts.on(
+        '-C', '--puppet-config-file CONFIGFILE',
+        'Puppet configuration file. Defaults to',
+        DEFAULT_PUPPET_CONF_FILE
+      ) do |puppet_conf_file|
+        @options[:puppet_conf_file] = puppet_conf_file
+      end
+
+      opts.on(
+        '-l', '--log-file LOGFILE',
+        'Log file. Defaults to',
+        DEFAULT_LOG_FILE
+      ) do |log_file|
+        @options[:log_file] = log_file
+      end
+
+      opts.on(
+        '-q', '--quiet',
+        'Quiet console output. When enabled,',
+        'console messages indicating progress are',
+        "suppressed. Defaults to #{DEFAULT_QUIET}."
+      ) do
+        @options[:quiet] = true
+      end
+
+      opts.on(
+        '-d', '--debug',
+        'Log processing details to the console.',
+        'These details are always written to the',
+        "log file. Defaults to #{DEFAULT_DEBUG}."
+      ) do
+        @options[:debug] = true
+      end
+
+      opts.on( '-h', '--help', 'Print this help message') do
+        @options[:help_requested] = true
+        puts opts
+      end
+    end
+
+    begin
+      opt_parser.parse!(args)
+      return if @options[:help_requested]
+
+      validate_options
+
+      cmd = [
+        '/opt/puppetlabs/bin/puppet',
+        'agent',
+        '--onetime',
+        '--no-daemonize',
+        '--no-show_diff',
+        '--no-splay',
+        '--verbose',
+        '--logdest', @options[:log_file],
+        '--waitforcert', @options[:puppet_wait_for_cert]
+      ]
+
+      cmd += [ '--evaltrace', '--summarize' ] if @options[:print_stats]
+      @options[:puppet_cmd] = cmd.join(' ')
+      reset_retry_interval
+
+    rescue OptionParser::ParseError => e
+      raise ConfigurationError.new("#{e.message}\n#{opt_parser.banner}")
+    end
+  end
+
+  def reset_retry_interval
+    @retry_interval = nil
+  end
+
+  def run(args)
```

Comment:
Nit: Could we make this method last in the class so that it's easier to find when debugging?